src/analysis/special_functions/stirling.lean

```diff
@@ -0,0 +1,237 @@
+/-
+Copyright (c) 2022. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Moritz Firsching, Fabian Kruse, Nikolas Kuhn
+-/
+import analysis.p_series
+import analysis.special_functions.log.deriv
+
+/-!
+# Stirling's formula
+
+This file proves Theorem 90 from the [100 Theorem List] <https://www.cs.ru.nl/~freek/100/>.
+It states that $n!$ grows asymptotically like $\sqrt{2\pi n}(\frac{n}{e})^n$.
+TODO: Add Part 2 to complete the proof
+
+## Proof outline
+
+The proof follows: <https://proofwiki.org/wiki/Stirling%27s_Formula>.
+
+### Part 1
+We consider the fraction sequence $a_n$ of fractions $\frac{n!}{\sqrt{2n}(\frac{n}{e})^n}$ and
+prove that this sequence converges against a real, positive number $a$. For this the two main
+ingredients are
+ - taking the logarithm of the sequence and
+ - use the series expansion of $\log(1 + x)$.
+-/
+
+open_locale topological_space big_operators
+open finset filter nat real
+
+namespace stirling
+
+/-!
+ ### Part 1
+ https://proofwiki.org/wiki/Stirling%27s_Formula#Part_1
+-/
+
+/--
+Define `stirling_seq n` as $\frac{n!}{\sqrt{2n}/(\frac{n}{e})^n$.
+Stirling's formula states that this sequence has limit $\sqrt(π)$.
+-/
+noncomputable def stirling_seq (n : ℕ) : ℝ :=
+n.factorial / (sqrt (2 * n) * (n / exp 1) ^ n)
+
+/-- Define `log_stirling_seq n` as the log of `stirling_seq n`. -/
+noncomputable def log_stirling_seq (n : ℕ) : ℝ := log (stirling_seq n)
```

Comment:
Do you suggest to replace all ~40 occurrences of `log_stirling_seq`? I think it would make quite a few things more cumbersome, so I just want to double check.

Reply:
I've tried it and it is really not hard: replace `log_stirling_seq x` by `log (stirling_seq x)`, the antitone lemma becomes `lemma log_stirling_seq'_antitone : antitone (log ∘ stirling_seq ∘ succ) :=` and remove one useless `log_stirling_seq` in some simp only. I don't care about the `log_stirling_seq'` since that is in a proof.

Reply:
Cool, I have done as you suggested!